README.md

```diff
@@ -15,11 +15,30 @@ Please ensure all tasks have a description that includes a clear definition of w
 
 Builds are run automatically out of this repository. Basic workflow is:
 
-- Create a branch and implement your changes
-- The Docker image is built on the branch and as part of the pull request
-- Once the PR is merged, a build will be run and pushed to `latest` and `sudo-latest` from master
-- When a new stable image is ready, create a release with a version number like `0.0.6` and
-this will be pushed to `0.0.6` and `sudo-0.0.6`.
+- Create a branch and implement your changes.
+- The Docker image is built on the branch and as part of the pull request.
+- The integration test will be run on PR, merge and release events.
+- Once the PR is merged, a build will run. The resulted image will be tagged with git `head` and `latest` and pushed to ECR.
+- When decide to release a new stable image, create a release with a version number with `major.minor.patch`, e.g.,  `2.0.1`.
+It will trigger the build and image will be tagged with `2.0.1` and `stable` and pushed to ECR.
+
+## Packages' version maintenance and upgrade
+
+The base environment is using conda and docker image is built in two stages:
+
+1. Create conda env and install as many as possible packages from `conda-forge`. Then `pip install` the rest, e.g., most `odc-` packages.
+2. Copy the conda env to a new ubuntu image.
+
+To speed up the building, the build workflow pulls the cache stored on ECR. Though with every build, the cache layers starting from `pip install` will be busted such that the newest versions of `odc-` packages will be installed. Thus, for version upgrade on these packages, creating a release will be enough.
+
+The old conda env cache is used for all builds unless `env.yml` is changed. Other than the reason of image building time cited above, it takes more effort to resolve the conflicts and maintain the code base of `odc-` when the major upgrade on some base packages happens, e.g., some geospatial related packages, `GDAL` and `GEOS`. Hence, after evaluate pros and cons, we decide to keep this part rather manual.
```

Comment:
I'm not sure whether I butchered this one.
```suggestion
The old Conda env cache is used for all builds unless `env.yml` is changed. In addition to speeding up builds, this cached environment allows us to maintain a working `odc-` codebase and defer resolving conflicts on geospatial base packages such as `GDAL` and `GEOS`, until we have a good opportunity to manually review them. While not ideal, this approach was adopted as the best compromise for our use case.
```

Reply:
I've observed some fun linguistic facts here. I use context aware expression subconsciously. "Other than..." means "besides, in addition to" within the context, but the full sentence is "I don't really have a good excuse of using conda cache other than the time reason mentioned above, but I do prefer solving the conflicts less frequently since a) it takes effort when happens, b)it doesn't seem necessary so far". And I'm not sure if it's either ideal or the best compromise, it's only a decision based on the current preference.

Reply:
Ahh 😄
And for whether or not the option was the best compromise, I think I read it as more conclusive from the `after evaluate pros and cons, we decide` bit.